crates/assertion-executor/src/db/version_db.rs

```diff
@@ -0,0 +1,701 @@
+    dont_read_from_inner_db: bool,
+}
+
+impl StorageEntry {
+    fn mark_selfdestructed(&mut self) {
+        self.dont_read_from_inner_db = true;
+        self.map.clear();
+    }
+}
+
+impl DatabaseTypes {
+    fn get_basic(&self, address: Address) -> Option<AccountInfo> {
+        self.basic.get(&address).map(|guard| guard.value().clone())
+    }
+
+    fn get_code(&self, code_hash: &B256) -> Option<Bytecode> {
+        self.code_by_hash
+            .get(code_hash)
+            .map(|guard| guard.value().clone())
+    }
+
+    fn get_storage(&self, address: Address, slot: U256) -> Option<B256> {
+        self.storage
+            .get(&address)
+            .and_then(|entry| entry.map.get(&slot).map(|guard| *guard.value()))
+    }
+
+    fn is_selfdestructed(&self, address: Address) -> bool {
+        self.storage
+            .get(&address)
+            .is_some_and(|entry| entry.dont_read_from_inner_db)
+    }
+
+    fn get_block_hash(&self, number: u64) -> Option<B256> {
+        self.block_hash.get(&number).map(|guard| *guard.value())
+    }
+
+    fn insert_basic(&mut self, address: Address, info: AccountInfo) {
+        self.basic.insert(address, info);
+    }
+
+    fn insert_code(&mut self, code_hash: B256, code: Bytecode) {
+        self.code_by_hash.insert(code_hash, code);
+    }
+
+    fn insert_storage(&mut self, address: Address, slot: U256, value: B256) {
+        let entry = if let Some(entry) = self.storage.get_mut(&address) {
+            entry
+        } else {
+            self.storage.insert(address, StorageEntry::default());
+            self.storage.get_mut(&address).expect("entry must exist")
+        };
+        entry.map.insert(slot, value);
+    }
+
+    fn insert_selfdestructed(&mut self, address: Address) {
+        let mut entry = if let Some(entry) = self.storage.get_mut(&address) {
+            entry
+        } else {
+            self.storage.insert(address, StorageEntry::default());
+            self.storage.get_mut(&address).expect("entry must exist")
+        };
+        entry.mark_selfdestructed();
+    }
+
+    #[allow(dead_code)]
+    fn insert_block_hash(&mut self, number: u64, block_hash: B256) {
+        self.block_hash.insert(number, block_hash);
+    }
+
+    fn apply_delta(&mut self, delta: &DatabaseTypes) {
+        for entry in &delta.basic {
+            self.basic.insert(*entry.key(), entry.value().clone());
+        }
+        for entry in &delta.code_by_hash {
+            self.code_by_hash
+                .insert(*entry.key(), entry.value().clone());
+        }
+        for entry in &delta.storage {
+            let address = *entry.key();
+            let delta_entry = entry.value();
+
+            let mut storage_entry = if let Some(entry) = self.storage.get_mut(&address) {
+                entry
+            } else {
+                self.storage.insert(address, StorageEntry::default());
+                self.storage.get_mut(&address).expect("entry must exist")
```

Comment:
```suggestion
let entry = self.storage.entry(address).or_insert_with(StorageEntry::default);
```

it applies to all occurrences